Add getting started documents for IDP (#1996)
harness/developer-hub: docs/internal-developer-portal/getting-started/enable-a-new-plugin.md

---
title: Enable a plugin in Harness IDP
description: Learn how you to enable a new plugin in Harness IDP and how to make use of it in your Software Catalog
sidebar_position: 50
helpdocs_topic_id:
helpdocs_category_id:
helpdocs_is_private: false
helpdocs_is_published: true
---

> Enabling plugins in Harness IDP allows you to extend its functionality and customize it to suit your specific needs. This guide will provide step-by-step instructions on how to enable a new plugin in Harness IDP.

Author: Yes - plugins extend IDP's functionality. Thanks!

- A plugin with both cards and tabs (multiple components) - e.g.
- (Optional) A plugin with both UI and backend parts
- A plugin with both UI and backend as well as a DB e.g. TODO plugin
- A full page plugin - e.g. GraphiQL

These are just a few examples of plugin types available in Harness IDP. Depending on your requirements, you can choose the plugin type that best aligns with your needs and enables the desired functionality in your IDP.

**Next Steps:**

Once you have selected the desired plugin type, you can proceed with enabling the plugin following the specific instructions provided by Harness IDP. The enabling process may involve configuration, activation, and updating layout steps which will vary depending on the plugin.

**Process:**

To enable a plugin, users can follow these steps:

1. All the admin configurations are provided in the Admin panel. Once users with admin permissions are onboarded, they can access the Admin option in the side nav. Clicking on the Admin section will navigate the user to the Admin view, where they can see multiple options such as Plugins, Layout, Connectors, Configurations, Access Control, and more.

![](static/plugin-page-nav.png)

> 2. The "Plugins" navigation option will direct the user to the plugins list page, which showcases a collection of curated plugins. On this page, there are distinct sections dedicated to both enabled and disabled plugins, providing a clear differentiation between the two categories.

Author: Got it, I'll adjust the numberings.

![](static/att_4_for_21398290667.png)

> 3. In the Plugins navigation, click on the desired plugin card. This will redirect users to the Plugin details screen, which provides information about the plugin such as its name, details, plugin layout, configurations, and the option to add a plugin secret.

Reviewer: ```suggestion
3. Select the desired plugin card. 
The Plugin details page is displayed. The page provides information such as the plugin's name, details, layout, and configurations. It also includes an option to add a plugin secret. The plugin's layout section is read-only and displays the exported cards, tabs, or pages provided by the plugin.
```

![](static/att_5_for_21398290667.png)

![](static/att_3_for_21398290667.png)

> 4. The plugin's layout section is read-only and displays the exported cards, tabs, or pages provided by the plugin.

Reviewer: Not a step. Merged into the previous step's info section. Remove and renumber.

> 5. In the configuration section, a default config YAML is provided, which can be edited using the YAML editor. Users have the option to customize the configuration based on their requirements.

Reviewer: ```suggestion
5. (Optional) If you want to customize the plugin's configuration, in the **Configurations** section, edit the default configuration YAML. 
```

> 6. Below the configuration, users can update the plugin secret token value as specified in the config YAML. Make sure the secret token name is the same as given in the config YAML.

Reviewer: ```suggestion
6. Update the plugin secret token value as specified in the configuration YAML. Make sure that the secret token name is the same as given in the configuration YAML.
```

7. After verifying all the configurations and secret details, users can save the configuration. If the config YAML is valid, the plugin configuration will be saved successfully. Otherwise, appropriate error messages will be displayed.

> 8. Upon saving the configuration, the "Enable Plugin" button will automatically be enabled. Users can enable the plugin by clicking on this button.

Reviewer: ```suggestion
8. To enable the plugin, select **Enable Plugin**. 
The plugin is moved to the Enabled Plugins section on the Plugins page.
```

> 9. Once the selected plugin is enabled, it will be moved to the "Enabled Plugins" section on the Plugins List page. Users can navigate to this page by clicking on the "Back" button.

Reviewer: Remove the first sentence. I added this sentence to the additional info section of the previous step. 
```suggestion
9. To return to the Plugins page, select **Back**.
```

> 10. After enabling the plugin, users are required to ingest the layout using plugin layout management. The layout should be modified according to the exported elements from the plugin, which may include cards, tabs, and pages.

Reviewer: ```suggestion
10. Ingest the layout by using plugin layout management. The layout should be modified according to the elements that are exported from the plugin. The elements might include cards, tabs, and pages.
```

> For instance, if the Jenkins plugin exports one card and one tab, in order to display them in your catalog entity, the user must update the layout accordingly. To add a dedicated tab for the Jenkins plugin, follow these steps:

Reviewer: ```suggestion
For instance, if the Jenkins plugin exports one card and one tab, in order to display them in your catalog entity, you must update the layout accordingly. To add a dedicated tab for the Jenkins plugin, follow these steps:
```

- From the dropdown menu, select the domain option.
- Search for the "ci-cd" component within the domain.

> - Inside the Entity Switch case, insert the following code snippet.

Reviewer: ```suggestion
- Inside the **Entity Switch** case, insert the following code snippet:
```

![](static/layout-snippet.png)

> These are the simple steps to enable a plugin in Harness IDP. Finally, the user will be able to view the enabled plugins in the catalog section of Harness IDP. To see the enabled plugins, the user simply needs to register the plugin catalog info YAML.

Reviewer: ```suggestion
You should now be able to view the enabled plugins in the catalog section of Harness IDP. To see the enabled plugins, register the plugin catalog info YAML.
```
